src/go/nerdctl-stub/command_handlers.go

```diff
@@ -24,13 +36,48 @@ func imageBuildHandler(c *commandDefinition, args []string) (*parsedArgs, error)
 	}
 	newPath, cleanups, err := filePathArgHandler(args[0])
 	if err != nil {
-		for _, cleanup := range cleanups {
-			cleanupErr := cleanup()
-			if cleanupErr != nil {
-				log.Printf("Error cleaning up: %s", cleanupErr)
-			}
-		}
+		runCleanups(cleanups)
 		return nil, err
 	}
 	return &parsedArgs{args: append([]string{newPath}, args[1:]...), cleanup: cleanups}, nil
 }
+
+// containerCopyHandler handles `nerdctl container cp`
+func containerCopyHandler(c *commandDefinition, args []string) (*parsedArgs, error) {
+	var resultArgs []string
+	var cleanups []cleanupFunc
+
+	// Positional arguments `nerdctl container cp` are all paths, whether inside
+	// the container or outside.
+
+	for _, arg := range args {
+		if arg == "-" {
+			// flag for stdin/stdout; don't need to deal with that.
+			resultArgs = append(resultArgs, arg)
+			continue
+		}
+		// There are three possible cases for the arg:
+		// - There are no `:`s in the string -- a host-side relative path.
+		// - Any single character before `:` -- assume a drive letter.
+		// - Any other sequence before `:` -- container specification.
+		// Note that the latter case means a container with a single-letter name
+		// will not correctly; but that's probably acceptable for now.
```

Comment:
Now, we could go further to disambiguate `nerdctl cp a:foo c:bar` by treating `src` as a container file only when `a` is a valid container id, e.g. by verifying that `nerdctl inspect a` runs without error.[^1]

This will only add overhead when both `src` and `dest` have a single character container/drive prefix, so should be acceptable, but I don't know how much extra effort this would be, and it may still end up guessing wrong anyways.

[^1]: This is still not guaranteed to match user intention, as drive letters `a` to `f` can still be interpreted as an abbreviated id instead of a name, so a container `a` can exist, even if the user meant it to specify a drive. That's why I think it may not be worth the additional complexity.

Reply:
> The `src` contains a `\`, or

For… reasons, `/` often works as a path separator on Windows too. There's also drive-relative paths (e.g. `C:foo` means "the file `foo` on drive `C`, in that drive's current working directory").

For that reason, I would propose that:
- If either argument starts with multiple characters before `:`, treat that as the container ID (and therefore the other one isn't).
- If we have a single-character drive-like-thing and one without a colon at all, assume the former is a container ID.
- If we have multiple single-character drive-like-things, check if those are drives (by `stat`ing `C:\` etc.) and assume that the one that fails (if any) is the container ID.
- If both succeed, assume the first (`src`) is the container ID.

This means that in certain cases (`nerdctl cp c:foo c:bar`), the user will need to change their command line to use the container ID (at least more characters of it) instead. Does that sound reasonable?

Reply:
> > The `src` contains a `\`, or
> 
> For… reasons, `/` often works as a path separator on Windows too. There's also drive-relative paths (e.g. `C:foo` means "the file `foo` on drive `C`, in that drive's current working directory").

I know. This is only used as a tie-breaker, when both `src` and `dest` have a single character container id, and you want the `src` to be the local file.

> For that reason, I would propose that:
> 
> * If either argument starts with multiple characters before `:`, treat that as the container ID (and therefore the other one isn't).
> * If we have a single-character drive-like-thing and one without a colon at all, assume the former is a container ID.

This matches the rules I've given above.

> * If we have multiple single-character drive-like-things, check if those are drives (by `stat`ing `C:\` etc.) and assume that the one that fails (if any) is the container ID.

This rule is not worth the effort, I think, because it can still be wrong. Just because you have a `C:` drive doesn't mean the user didn't mean container `C12DE34` when they wrote `C:file`.

> * If both succeed, assume the first (`src`) is the container ID.

That again is the rule above.
 
> This means that in certain cases (`nerdctl cp c:foo c:bar`), the user will need to change their command line to use the container ID (at least more characters of it) instead. Does that sound reasonable?

I fail to see how your description is different from what I described above (with the exception of the `stat` call, which I think isn't worth it).